src/neon/intrinsics.rs

```diff
@@ -0,0 +1,395 @@
+//use std::arch::
+
+use crate::neon::simd_llvm;
+
+use std::mem;
+use std::hint::unreachable_unchecked;
+use core;
+
+#[allow(unused)]
+macro_rules! types {
+    ($(
+        $(#[$doc:meta])*
+        pub struct $name:ident($($fields:tt)*);
+    )*) => ($(
+        $(#[$doc])*
+        #[derive(Copy, Clone, Debug)]
+        #[allow(non_camel_case_types)]
+        #[repr(simd)]
+        #[allow(clippy::missing_inline_in_public_items)]
+        pub struct $name($($fields)*);
+    )*)
+}
+
+extern "C" {
+    #[link_name = "llvm.aarch64.neon.pmull64"]
+    fn vmull_p64_(a: i64, b: i64) -> int8x16_t;
+    #[link_name = "llvm.aarch64.neon.vshrq"]
+    fn vshrq_n_u8_(a: uint8x16_t, b: u8) -> uint8x16_t;
+    #[link_name = "llvm.aarch64.neon.addp.v16i8"]
+    fn vpaddq_u8_(a: uint8x16_t, b: uint8x16_t) -> uint8x16_t;
+    #[link_name = "llvm.aarch64.neon.addp.v16i8"]
+    fn vaddq_s8_(a: int8x16_t, b: int8x16_t) -> int8x16_t;
+    #[link_name = "llvm.aarch64.neon.addp.v16i32"]
+    fn vaddq_s32_(a: int32x4_t, b: int32x4_t) -> int32x4_t;
+    #[link_name = "llvm.aarch64.neon.vextq.v16s8"]
+    fn vextq_s8_(a: int8x16_t, b: int8x16_t, n:u8) -> int8x16_t;
+    #[link_name = "llvm.aarch64.neon.vtstq.v16u8"]
+    fn vtstq_u8_(a: uint8x16_t, b: uint8x16_t) -> uint8x16_t;
+    #[link_name = "llvm.aarch64.neon.vtstq.v16s8"]
+    fn vtstq_s8_(a: int8x16_t, b: int8x16_t) -> int8x16_t;
+    #[link_name = "llvm.ctpop.u64"]
+    fn ctpop_u64_(a: u64) -> u32;
+    #[link_name = "llvm.ctlz.u64"]
+    fn ctlz_u64_(a: u64) -> u32;
+    #[link_name = "llvm.cttz.u64"]
+    fn cttz_u64_(a: u64) -> u32;
+}
```

Comment:
haven't tested many of the above ... not completely sure they're right (or even close)

Reply:
Do we need those? I went over the simdjson code and tried to find all the intrinsics used but I might have missed some / or things changed. If any are missing you can add them to the list in #32 and I'll add them :)

Reply:
@Licenser I think so, unless there's equivalent simd_llvm idiom for hamming() and trailingzeroes()?